nasa-pictures-stealer/src/main/java/nasapicturesstealer/infrastructure/repository/NasaPicturesRepository.java

```diff
@@ -0,0 +1,72 @@
+package nasapicturesstealer.infrastructure.repository;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import lombok.RequiredArgsConstructor;
+import nasapicturesstealer.domain.entity.NasaCamera;
+import nasapicturesstealer.domain.entity.NasaPicture;
+import nasapicturesstealer.infrastructure.entity.NasaCameraEntity;
+import nasapicturesstealer.infrastructure.entity.NasaPictureEntity;
+import nasapicturesstealer.infrastructure.mapper.NasaCameraMapper;
+import nasapicturesstealer.infrastructure.mapper.NasaPictureMapper;
+import org.springframework.data.util.Pair;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class NasaPicturesRepository {
+
+    private final NasaPictureEntityRepository nasaPictureEntityRepository;
+  
+    private final NasaCameraEntityRepository nasaCameraEntityRepository;
+    
+    private final NasaPictureMapper nasaPictureMapper;
+    
+    private final NasaCameraMapper nasaCameraMapper;
+    
+    public List<NasaPicture> saveDomainEntities(List<NasaPicture> pictures) {
+      Map<Integer, Long> nasaIdToPictureId = this.nasaPictureEntityRepository
+        .findAllByNasaIds(pictures.stream().map(NasaPicture::getNasaId).toList())
```

Comment:
The overall length of the SQL query is limited. PostgreSQL does not have a hard limit, but I have already encountered such a problem on several projects. Such requests should be made only in batches